projects/sampling/component.py

```diff
@@ -0,0 +1,122 @@
+"""Undersampling component definition."""
+
+from typing import Optional, Text, List
+from executor import Executor
```

Comment:
Circular dependency here, `from exceutor import Executor` and then in `executor` import `component`. This dependency only seems to exist because of all the keys you define. This only seems to affect me when I try deploying to Kubeflow, doesn't seem to be an issue when running locally on my machine. My fix was extracting the keys into a separate file i.e. `keys.py` and then importing within `component` and `executor`.